### DISCUSSION TITLE
Performance Tracing with Shrink-Expand-Mechanism

### DISCUSSION THREAD

**Discussion by manubiwa** [UIUC-PPL/projections#143] [General]
Hi everyone!

I work with the shrink-expand mechansim of charm++. I wanted to test the projections tool with the build option --enable-tracing and the compile option -tracemode projections. Because of these options the charm++ application doesn't run anymore. Are there ways to use the projections tool with the shrink-expand mechanism? Are there other compile or build options?

**Comment by rbuch** [comment]
I don't think there's anything that's fundamentally incompatible with using Projections with shrink-expand. Can you provide some specifics on what errors you are seeing?

**Comment by manubiwa** [comment]
I work with following:
Virtual machine with Ubuntu 20.04.4 LTS
Build charm++ with  "./build charm++ netlrts-linux-x86_64 --enable-tracing --enable-shrinkexpand"
Compiled the /examples/charm++/shrink_expand/jacobi2d-iter program with the option "-tracemode projections"
Started the program with "./charmrun +p4 jacobi2d 200 20 +balancer GreedyLB +LBDebug 3 ++server ++server-port 1234"

After starting the program an error appears:

------------- Processor 3 Exiting: Called CmiAbort ------------
Reason: registerExitFn is called when shrink-expand is enabled!
Charm++> scheduler running in netpoll mode.
[3] Stack Traceback:
  [3:0] _Z14CmiAbortHelperPKcS0_S0_ii+0xbd  [0x55612163608d]
  [3:1] CmiGetNonLocal+0  [0x5561216360cc]
  [3:2] +0x2af70f  [0x5561214cd70f]
  [3:3] _Z23initTraceProjectionsBOCv+0x14  [0x55612167d755]
  [3:4] _ZN13InitCallTable18enumerateInitCallsEv+0x47  [0x5561214cbfe9]
  [3:5] _Z10_initCharmiPPc+0x677  [0x5561214cc81a]
  [3:6] +0x417e4a  [0x556121635e4a]
  [3:7] ConverseInit+0x658  [0x556121635d39]
  [3:8] charm_main+0x45  [0x5561214cd6c1]
  [3:9] main+0x24  [0x5561214c9d3f]
  [3:10] __libc_start_main+0xf3  [0x7fbce6ce50b3]
  [3:11] _start+0x2e  [0x55612146dc1e]
Fatal error on PE 3> registerExitFn is called when shrink-expand is enabled!

**Comment by rbuch** [comment]
Aha, thanks for the report, that certainly looks like a problem, we'll investigate a bit and get back to you.

**Comment by rbuch** [comment]
I've created an issue from your report here: https://github.com/UIUC-PPL/charm/issues/3613 (the problem is on the Charm++ side rather than with Projections itself, which is why it's in the `charm` repo).

**Reply by manubiwa** [reply]
Thanks!
